src/AzureClient/AzureClient.cs

```diff
@@ -1,24 +1,249 @@
+            if (string.IsNullOrEmpty(cacheDirectory))
+            {
+                cacheDirectory = Path.Join(System.Environment.GetFolderPath(System.Environment.SpecialFolder.UserProfile), ".azure-quantum");
+            }
+
+            var storageCreationProperties = new StorageCreationPropertiesBuilder(cacheFileName, cacheDirectory, clientId).Build();
+            var cacheHelper = await MsalCacheHelper.CreateAsync(storageCreationProperties);
+            cacheHelper.RegisterCache(msalApp.UserTokenCache);
+
+            var scopes = new List<string>() { "https://quantum.microsoft.com/Jobs.ReadWrite" };
+
+            bool promptForLogin = forceLogin;
+            if (!promptForLogin)
+            { 
+                try
+                {
+                    var accounts = await msalApp.GetAccountsAsync();
+                    AuthenticationResult = await msalApp.AcquireTokenSilent(
+                        scopes, accounts.FirstOrDefault()).WithAuthority(msalApp.Authority).ExecuteAsync();
+                }
+                catch (MsalUiRequiredException)
+                {
+                    promptForLogin = true;
+                }
+            }
+
+            if (promptForLogin)
+            {
+                AuthenticationResult = await msalApp.AcquireTokenWithDeviceCode(scopes,
+                    deviceCodeResult =>
+                    {
+                        channel.Stdout(deviceCodeResult.Message);
+                        return Task.FromResult(0);
+                    }).WithAuthority(msalApp.Authority).ExecuteAsync();
+            }
+
+            if (AuthenticationResult == null)
+            {
+                return AzureClientError.AuthenticationFailed.ToExecutionResult();
+            }
+
+            var credentials = new Rest.TokenCredentials(AuthenticationResult.AccessToken);
+
+            QuantumClient = new QuantumClient(credentials);
+            QuantumClient.SubscriptionId = subscriptionId;
+            QuantumClient.ResourceGroupName = resourceGroupName;
+            QuantumClient.WorkspaceName = workspaceName;
+
+            ActiveWorkspace = new Azure.Quantum.Workspace(
+                QuantumClient.SubscriptionId, QuantumClient.ResourceGroupName,
+                QuantumClient.WorkspaceName, AuthenticationResult?.AccessToken);
+
+            try
+            {
+                var jobsList = await QuantumClient.Jobs.ListAsync();
+                channel.Stdout($"Found {jobsList.Count()} jobs in Azure Quantum workspace {workspaceName}");
+            }
+            catch (Exception e)
+            {
+                channel.Stderr(e.ToString());
+                return AzureClientError.WorkspaceNotFound.ToExecutionResult();
+            }
+
+            return QuantumClient.ToJupyterTable().ToExecutionResult();
+        }
+
+        /// <inheritdoc/>
+        public async Task<ExecutionResult> PrintConnectionStatusAsync(IChannel channel)
```

Comment:
This could probably be an expression body method, using something like `QuantumClient == null ? AzureClientError.NotConnected.ToExecutionResult() :  QuantumClient.ToJupyterTable().ToExecutionResult()`.